bluetooth/update/README.md

```diff
@@ -2,145 +2,133 @@
+           FROM gis.centreline_intersection z
+          ORDER BY (z.geom <-> mohan.new_added_detectors.from_geom)
+         LIMIT 1) nodes);
 ```
-Draws lines using the reader locations. This is the result.
+Check that correct intersections are returned from this query especially for oblique intersections with an offset. If required, correct the intersection_id and geom for such intersections and finalize the table `mohan.bluetooth_nodes`. An example is shown below. The intersection between Dundas St W, Roncesvalles Ave and Boustead Ave is oblique with an offset. The BT reader 5263 (red dot) picked up intersection_id `13466305` (greed dot) as nearest intersection to the reader. But the planned route does not go through this intersection_id. The correct intersection_id for this reader is `113466258`. The intersection_id and geom for this redear was therefore corrected manually.  
+ 
+
+![odd_shaped_intersections](img/odd_intersection.PNG)
 
-![bt_lines](img/bt_lines.PNG)
+## Using pg_routing
+Once the nearest centreline intersection nodes are linked to the bluetooth readers geom `mohan.bluetooth_nodes`, we are ready to run the following Query to create new routes. 
 
 ```SQL
-, conflation AS (
-SELECT analysis_id, 
-	corridor, 
-	from_intersection, 
-	from_long, 
-	from_lat, 
-	to_intersection, 
-	to_long, 
-	to_lat, s.geom, geo_id, s.lf_name, 
-	ST_Length(st_intersection(ST_BUFFER(ST_Transform(ST_SetSRID(line, 4326), 32190), 1.5*b.length, 'endcap=flat join=round') , ST_Transform(s.geom, 32190))) /ST_Length(ST_Transform(s.geom, 32190))
-	
-	FROM make_lines b
-
-INNER JOIN rliu.bt_centreline s 
-ON 
-ST_DWithin( ST_Transform(s.geom, 32190) ,  ST_BUFFER(ST_Transform(ST_SetSRID(line, 4326), 32190), 3*b.length, 'endcap=flat join=round') , 1000)
-
-
---AND ST_Length(st_intersection(ST_BUFFER(ST_Transform(ST_SetSRID(line, 4326), 32190), 1.5*b.length, 'endcap=flat join=round') , ST_Transform(s.geom, 32190))) /ST_Length(ST_Transform(s.geom, 32190)) > 0.4
+CREATE table mohan.bt_segments_new AS (
+WITH lookup AS (
+SELECT analysis_id, from_id, origin.int_id AS source, to_id, dest.int_id AS target
+FROM mohan.new_added_detectors 
+INNER JOIN mohan.bluetooth_nodes origin ON from_id = origin.bluetooth_id 
+INNER JOIN mohan.bluetooth_nodes dest ON to_id = dest.bluetooth_id
+),
+results AS (
+	SELECT * 
+	FROM lookup
+			 CROSS JOIN LATERAL pgr_dijkstra('SELECT id, source, target, cost FROM gis.centreline_routing_directional inner join gis.centreline on geo_id = id
+where fcode != 207001', source::int, target::int, TRUE)		 
+), 
+lines as (
+	SELECT analysis_id, street, direction, from_street, to_street, edge AS geo_id, geom 
+	FROM results			 
+INNER JOIN gis.centreline ON edge=geo_id
+INNER JOIN mohan.new_added_detectors USING (analysis_id)
+ORDER BY analysis_id
 )
-
-SELECT * 
-INTO rliu.new_bt_coflate1000 
-FROM conflation
+SELECT analysis_id, street, direction, from_street, to_street,
+	CASE WHEN geom_dir != direction THEN ST_reverse(geom) 
```

Comment:
not sure if this is needed because we are using centreline with direction as a base network for which the direction of the line is already flipped. This might result in double flipped geometries if the dirs didnt match because of a NB vs WB comparison